Fig.Core/Settings.cs

```diff
@@ -60,6 +60,27 @@ public bool Update(object val)
             }
         }
 
+        private interface IBindResult
```

Comment:
I'm curious why you chose an interface and not just going with the class as is.

Reply:
I found that the way I am using reflection and recursion, it would fail to be able to return the correct type for nested types.  Here's the line that was the reason for the change: https://github.com/rofr/Fig/blob/5c98dd4167e2c58bd11fe6a214a42cb844d482aa/Fig.Core/Settings.cs#L231